### PR TITLE
chore: Show tx error in toast

### PR DESCRIPTION
**Pull request by chefilip**

### CODE REVIEW

File: src/hooks/useCatchTxError.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        handleNormalError(error)
+      } else {
+        library
+          .call(tx, tx.blockNumber)
+          .then(() => {
+            handleNormalError(error)
+          })
+          .catch((err: any) => {
+            // -32000 is insufficient funds for gas * price + value
+            if (err?.data?.code === -32000) {
```
Comment (chef-jojo) [inline src/hooks/useCatchTxError.ts:48]: Let's make it a function, similar to 
https://github.com/pancakeswap/pancake-frontend/blob/develop/src/utils/sentry.ts#L20-L23

File: src/views/Pools/components/CakeVaultCard/VaultStakeModal.tsx [hunk 2]
```diff
@@ -169,66 +182,39 @@ const VaultStakeModal: React.FC<VaultStakeModalProps> = ({
...
-      try {
-        const tx = await callWithGasPrice(
-          vaultPoolContract,
-          'withdraw',
-          [shareStakeToWithdraw.sharesAsBigNumber.toString()],
-          callOptions,
-        )
+      },
+      () => tx,
+      () => setPendingTx(false),
```
Comment (memoyil) [inline src/views/Pools/components/CakeVaultCard/VaultStakeModal.tsx:187]: setPendingTx in line 181 can be removed

File: src/hooks/useCatchTxError.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+export default function useCatchTxError(): CatchTxErrorFunction {
+  const { library } = useWeb3React()
+  const { t } = useTranslation()
+  const { toastError } = useToast()
+
+  const handleNormalError = useCallback(
+    (error) => {
+      logError(error)
+
+      toastError(t('Error'), t('Please try again. Confirm the transaction and make sure you are paying enough gas!'))
```
Comment (chef-jojo) [inline src/hooks/useCatchTxError.ts:34]: let's skip toast error when `isUserRejected`

File: src/hooks/useCatchTxError.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            handleNormalError(error)
+          } else {
+            library
+              .call(tx, tx.blockNumber)
+              .then(() => {
+                handleNormalError(error)
+              })
+              .catch((err: TxError) => {
+                if (isGasEstimationError(err)) {
+                  handleNormalError(error)
```
Comment (chef-jojo) [inline src/hooks/useCatchTxError.tsx:75]: we can use ToastDescriptionWithTx here, maybe handleNormalError can accept a optional tx

**Review by chef-jojo — APPROVED** [review]

File: src/hooks/useCatchTxError.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                  const isRevertedError = indexInfo >= 0
+
+                  if (isRevertedError) reason = reason.substr(indexInfo + REVERT_STR.length)
+
+                  toastError(
+                    'Failed',
+                    <ToastDescriptionWithTx txHash={tx.hash}>
+                      {isRevertedError
+                        ? `Transaction failed with error: ${reason}`
+                        : 'Transaction failed. For detail error message:'}
```
Comment (Chef-Cheems) [inline src/hooks/useCatchTxError.tsx:117]:
```suggestion
                        : 'Transaction failed. For detailed error message:'}
```

**Pull request merged**